### DISCUSSION TITLE
UI Tweak - Add loading indicator to live fields in forms

### DISCUSSION THREAD

**Discussion by Mrkbingham** [filamentphp/filament#12431] [Ideas]
When using forms that have [complex dependant logic](https://filamentphp.com/docs/3.x/forms/getting-started#dependant-fields), or when dealing with performance issues/latency while using `->live()` on form components, the end user can still access these fields while they are in an inaccurate state, causing confusion and complexity.  I'm not aware of any way to represent a loading indicator on individual fields.

I've scoured discord and this github repo and can't find anything - but if there's a better approach, happy to utilize that to achieve my goals individually in my app.  But I figured this update would be beneficial to everyone using Filament.

My proposal here is to add the loading indicator to the left-side of fields - adding a small amount of left-margin during the load.  Sample fork provided and sample video here @danharrin 

https://github.com/filamentphp/filament/assets/4382816/4a26d799-8838-4ab9-9656-c79ce0f755e0

Link to my forked repo's branch dedicated to just a sample change isolated to the Select component:
https://github.com/Mrkbingham/mark-bingham-filament/tree/forms_wire_loading_spinner

[Specifically, this commit shows the change](https://github.com/Mrkbingham/mark-bingham-filament/commit/25c0899f041193e116e03b032c33f3c225adefb8)

If this is a worthwhile change, I would add appropriate loading spinners to all form components.

**Comment by danharrin** [comment]
I think there would need to be some opt-in setting for this, where you MUST specify the names of other fields that should trigger the spinner (and then pass to wire:target) - otherwise for every LW request the spinner is shown even if its totally unrelated like the form being saved

**Reply by rvzug** [reply]
Isn't it preferable to have 1 spinner on a dedicated spot, so 1 input element doesn't trigger x-quantity of spinners? Like before the save button? I imagine 5 selects that are triggered by one input element. It could be a mess UX-wise...

**Comment by cloudenum** [comment]
Any news on this? because this would improve the perceived app performance.

**Reply by danharrin** [reply]
I would welcome a PR for v4

**Comment by Mrkbingham** [comment]
Just adding in an update on this one.  After migrating hosting providers, I've dramatically improved some network bottlenecks in my app.  Personally, I don't have much of a need for this anymore.  Just wanted to provide transparency on where I'm at, and likely won't pursue this as an option.  Agreed on the comments above that if anyone else decides to pick this up, it should definitely be configurable/target-able.
